Description:
 Depending on some factors, such as the build directory, the build of
 QtWebEngine may attempt to create file names which are over the 255 character
 limit on Linux. Ref: https://bugreports.qt.io/browse/QTBUG-59769
Origin: Attached to referenced bug as workaround
Last-Update: 2018-09-25

--- a/src/3rdparty/chromium/tools/gn/ninja_action_target_writer.cc
+++ b/src/3rdparty/chromium/tools/gn/ninja_action_target_writer.cc
@@ -115,9 +115,18 @@ std::string NinjaActionTargetWriter::WriteRuleDefinition() {
     // strictly necessary for regular one-shot actions, but it's easier to
     // just always define unique_name.
     std::string rspfile = custom_rule_name;
+
+    //quick workaround if filename length > 255 - ".rsp", just cut the dirs starting from the end
+    //please note ".$unique_name" is not used at the moment
+    int pos = 0;
+    std::string delimiter("_");
+    while (rspfile.length() > 251 && (pos = rspfile.find_last_of(delimiter)) != std::string::npos)
+        rspfile = rspfile.substr(0,pos);
+
     if (!target_->sources().empty())
       rspfile += ".$unique_name";
     rspfile += ".rsp";
+
     out_ << "  rspfile = " << rspfile << std::endl;
 
     // Response file contents.
